scripts/docker-release/build_docker.sh

```diff
@@ -12,18 +12,7 @@ elif ! docker version
 then
   echo "ERROR: Building Docker image requires Docker daemon to be running" && exit 1
 fi
-
-echo "INFO: Determining latest tag"
-if [ ! -z ${TAG_NAME+x} ]
-then
-  echo "INFO: Detected TAG_NAME variable. Probably a Jenkins instance."
-  readonly GIT_TAG_DEFAULT=$(echo $TAG_NAME|sed s/^v//)
-else
-  echo "INFO: Did not detect TAG_NAME. Examining git log for latest tag"
-  readonly GIT_TAG_DEFAULT=$(git describe --abbrev=0|sed s/^v//)
-fi
-
-readonly GIT_TAG=${GIT_TAG:-$GIT_TAG_DEFAULT}
+readonly GIT_TAG=${1}
```

Comment:
Looking at the script, this variable needs to actually contain the version name *without* the v-prefix.

This is also shown by the previous runs:
![image](https://github.com/elastic/apm-agent-java/assets/6066661/626dcae4-ac17-4ec4-9700-6f7ef067d63f)

This means you need to:
 * Change the input to the script to be the version and not the tag
 * Maybe rename the variable to avoid confusion?
 
 The same thing applies to `push_docker`

Reply:
this also confirms that the image was published without the `v` prefix, so it's not strictly the git tag but the version number: https://container-library.elastic.co/r/observability/apm-agent-java.

Reply:
Good catch, let's use `RELEASE_VERSION` instead